### ISSUE TITLE
Ask browser permission only on start recording

### ISSUE THREAD

**Issue by anshad** [0x006F/react-media-recorder#26]
Right now it is asking permission on page load and it disappoints the users when we implement this snippet on a large web app. 

So can we add a condition to ask permission only on start recording?
@0x006F

**Comment by 0x006F** [comment]
That was by design, since we're dealing with cams and mic. Just wanted to let the user know about we've a component that's going to access their media devices (just like video conf services).

Asking permissions on-demand won't hurt much, but we need to test it thoroughly so that we won't annoy the end-user also. An edge case (as I recall) would be to record the screen. Upon selecting which window to share the browser, it'll switch immediately to that shared window which _may_ break the app's workflow. 

If I may, why your users are getting annoyed with the permissions dialog?

**Comment by arieller** [comment]
This is an issue for development as well, because on re render the permissions are getting asked over and over without actually helping the utility. Is there any way to remove the permission asking on mount ?

**Comment by neilnahid** [comment]
I second this.

**Comment by arieller** [comment]
> I second this.

Personally, I removed the initial permission ask on mount from the source code.

**Comment by nicgirault** [comment]
I was expecting to be able to set `audio: false` and switch it to true when I need the permission to be granted but having `audio: false` generates a `no_contraints` error. I would suggest to handle `audio: false` input to fix this issue

**Comment by sagodi97** [comment]
Yes,  @anshad,  if recording audio/video is a tiny part of a large web app, the user is more likely to decline access on load, because he/she won't know right away what is it for.

However, if this is by design you can get around it by checking permissions on mount. Something like:

```typescript
  const [audioAuth, setAudioAuth] = React.useState(false);
  React.useEffect(() => {
    if (!audioAuth) {
      navigator.permissions.query({ name: "microphone" }).then(({ state }) => {
        if (state === "granted") {
          setAudioAuth(true);
        }
      });
    }
  }, []);
```

**Comment by 0x006F** [comment]
This should be fixed in #57 . New version `1.5.0`
